### PR TITLE
Adding benchmark chart to readme

### PR DESCRIPTION
**Pull request by wduraes**

### CODE REVIEW

File: sdk/docs/iot/README.md [hunk 1]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
  Retry Policies | &radic;* | The Azure SDK for Embedded C provides guidelines for retries, but actual retries should be handled by the application.
  [IoT Plug and Play](https://docs.microsoft.com/azure/iot-pnp/overview-iot-plug-and-play) | &radic; | IoT Plug and Play enables solution builders to integrate smart devices with their solutions without any manual configuration.
 
+<br>
+
+## Size Chart
+
+The following chart shows the RAM and ROM usage for the PIC24-IOT dev kit from Microchip found in https://github.com/Azure-Samples/Microchip-PIC-IoT-Wx compiled using Microchip’s XC16 version 1.60 compiler with the following macros: AZ_NO_LOGGING, AZ_NO_CONDITION_CHECKING and NDEBUG.
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:78]: What is `NDEBUG`?

Also, did you mean `AZ_NO_PRECONDITION_CHECKING`?

The CMake options for those are as follows:
https://github.com/Azure/azure-sdk-for-c#cmake-options

```suggestion
The following chart shows the RAM and ROM usage for the PIC24-IOT dev kit from Microchip found in https://github.com/Azure-Samples/Microchip-PIC-IoT-Wx compiled using Microchip’s XC16 version 1.60 compiler with the following  compilation, preprocessor options: AZ_NO_LOGGING and AZ_NO_PRECONDITION_CHECKING.
```

File: sdk/docs/iot/README.md [hunk 2]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
...
+
+This PIC24-IOT kit has 2 MCUs (microcontrollers):
+ - Host MCU, a 16-bit **PIC24FJ128GA705** with 128 KB Flash Memory & 16 KB SRAM: this MCU runs the Embedded C SDK, the sample code and the MQTT client provided by Microchip.
+ - Wi-fi MCU, an **ATWINC**1510 Wi-Fi Module: this MCU runs the TLS and TCP stack. 
+
+The Pic24 sample includes both Hub and DPS services. The table below shows RAM/ ROM sizes considering:
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:84]:
```suggestion
The Pic24 sample includes both Hub and DPS services. The table below shows RAM/ROM sizes considering:
```

File: sdk/docs/iot/README.md [hunk 3]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
...
+-  Embedded C SDK libraries only – which represent the baseline size 
+-  Total size – which includes the Embedded C SDK and Microchip MQTT client (this Dev Kit has a separate Wi-Fi module which runs the TLS and TCP/IP stacks).
+
+|  | Embedded C SDK size | | Total Size | |
+|---------|----------|---------|---------|---------
+|**Sample** | **Program/ROM** | **Data/RAM** | **Program/ROM** | **Data/RAM** | 
+| PIC24 (Hub + DPS + IoT Plug and Play) | 26.15KB | 0 | 103.61KB | 10.57KB
+| PIC24 Telemetry only | 2.58KB | 0 | 74.16KB | 8.26KB
+
+<br>This table will be expanded as new Hardware samples become available.<br>
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:95]: Is the `<br>` necessary? The extra new line seems odd when rendered on GitHub.

```suggestion
This table will be expanded as new hardware samples become available.
```

File: sdk/docs/iot/README.md [hunk 4]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
...
+
+
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:97]:
```suggestion
```

File: sdk/docs/iot/README.md [hunk 5]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
...
  [IoT Plug and Play](https://docs.microsoft.com/azure/iot-pnp/overview-iot-plug-and-play) | &radic; | IoT Plug and Play enables solution builders to integrate smart devices with their solutions without any manual configuration.
 
+<br>
+
+## Size Chart
+
+The following chart shows the RAM and ROM usage for the PIC24-IOT dev kit from Microchip found in https://github.com/Azure-Samples/Microchip-PIC-IoT-Wx compiled using Microchip’s XC16 version 1.60 compiler with the following macros: AZ_NO_LOGGING, AZ_NO_CONDITION_CHECKING and NDEBUG.
+
+This PIC24-IOT kit has 2 MCUs (microcontrollers):
+ - Host MCU, a 16-bit **PIC24FJ128GA705** with 128 KB Flash Memory & 16 KB SRAM: this MCU runs the Embedded C SDK, the sample code and the MQTT client provided by Microchip.
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:81]: Add  a new line between text and list, to be consistent with below.
```suggestion

 - Host MCU, a 16-bit **PIC24FJ128GA705** with 128 KB Flash Memory & 16 KB SRAM: this MCU runs the Embedded C SDK, the sample code and the MQTT client provided by Microchip.
```

File: sdk/docs/iot/README.md [hunk 6]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
...
+The Pic24 sample includes both Hub and DPS services. The table below shows RAM/ ROM sizes considering:
+
+-  Embedded C SDK libraries only – which represent the baseline size 
+-  Total size – which includes the Embedded C SDK and Microchip MQTT client (this Dev Kit has a separate Wi-Fi module which runs the TLS and TCP/IP stacks).
+
+|  | Embedded C SDK size | | Total Size | |
+|---------|----------|---------|---------|---------
+|**Sample** | **Program/ROM** | **Data/RAM** | **Program/ROM** | **Data/RAM** | 
+| PIC24 (Hub + DPS + IoT Plug and Play) | 26.15KB | 0 | 103.61KB | 10.57KB
+| PIC24 Telemetry only | 2.58KB | 0 | 74.16KB | 8.26KB
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:93]: Just seeing absolute numbers, the question comes to mind: Are these numbers good? What's the goal or comparison to the alternative/previous, which would highlight the value of the new Embedded SDK?

File: sdk/docs/iot/README.md [hunk 7]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
...
 
+<br>
+
+## Size Chart
+
+The following chart shows the RAM and ROM usage for the PIC24-IOT dev kit from Microchip found in https://github.com/Azure-Samples/Microchip-PIC-IoT-Wx compiled using Microchip’s XC16 version 1.60 compiler with the following macros: AZ_NO_LOGGING, AZ_NO_CONDITION_CHECKING and NDEBUG.
+
+This PIC24-IOT kit has 2 MCUs (microcontrollers):
+ - Host MCU, a 16-bit **PIC24FJ128GA705** with 128 KB Flash Memory & 16 KB SRAM: this MCU runs the Embedded C SDK, the sample code and the MQTT client provided by Microchip.
+ - Wi-fi MCU, an **ATWINC**1510 Wi-Fi Module: this MCU runs the TLS and TCP stack.
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:82]:
```suggestion
 - Wi-fi MCU, an **ATWINC**1510 Wi-Fi Module: this MCU runs the TLS and TCP stack.
```

File: sdk/docs/iot/README.md [hunk 8]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
  Retry Policies | &radic;* | The Azure SDK for Embedded C provides guidelines for retries, but actual retries should be handled by the application.
  [IoT Plug and Play](https://docs.microsoft.com/azure/iot-pnp/overview-iot-plug-and-play) | &radic; | IoT Plug and Play enables solution builders to integrate smart devices with their solutions without any manual configuration.
 
+<br>
+
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:75]: This shouldn't be necessary. GitHub renders new lines fine. If you were using VSCode and its Markdown Preview, that doesn't handle tables well, and makes it seem like an extra new line is needed, causing the author to think they need to add more new lines like this. That ends up looking odd, due to the extra spacing:
image

```suggestion
```

File: sdk/docs/iot/README.md [hunk 9]
```diff
@@ -71,6 +71,30 @@ Feature | Azure SDK for Embedded C | Description
...
+The following chart shows the RAM and ROM usage for the PIC24-IOT dev kit from Microchip found in https://github.com/Azure-Samples/Microchip-PIC-IoT-Wx compiled using Microchip’s XC16 version 1.60 compiler with the following macros: AZ_NO_LOGGING, AZ_NO_CONDITION_CHECKING and NDEBUG.
+
+This PIC24-IOT kit has 2 MCUs (microcontrollers):
+ - Host MCU, a 16-bit **PIC24FJ128GA705** with 128 KB Flash Memory & 16 KB SRAM: this MCU runs the Embedded C SDK, the sample code and the MQTT client provided by Microchip.
+ - Wi-fi MCU, an **ATWINC**1510 Wi-Fi Module: this MCU runs the TLS and TCP stack. 
+
+The Pic24 sample includes both Hub and DPS services. The table below shows RAM/ ROM sizes considering:
+
+-  Embedded C SDK libraries only – which represent the baseline size
```
Comment (ahsonkhan) [inline sdk/docs/iot/README.md:86]:
```suggestion
-  Embedded C SDK libraries only – which represent the baseline size.
```

File: sdk/docs/iot/README.md (hunk 2 above)
Comment (momuno) [inline sdk/docs/iot/README.md:84]: I would reword to:

```suggestion
The chart considers the following two size comparisons:  

- Embedded C SDK size -  This refers only to the libraries of the Embedded C SDK.  It is the baseline size.
- Total size - This includes both the Embedded C SDK libraries and also the Microchip MQTT client, which has a separate Wi-Fi module running the TLS and TCP/IP stacks.
```

**Review by momuno — APPROVED** [review]

File: sdk/docs/iot/README.md (hunk 6 above)
Comment (wduraes) [reply sdk/docs/iot/README.md:92]: Comparison shows the memory usage when you have only telemetry x all features. It highlights how small of a footprint can be used in a device with the Embedded C SDK.
